icu4j/main/classes/core/src/com/ibm/icu/impl/number/DecimalQuantity.java

```diff
@@ -122,6 +122,13 @@
      */
     public int getMagnitude() throws ArithmeticException;
 
+    /**
+     * @return The value of {@link #getMagnitude} for the original input number.
+     * @throws ArithmeticException
+     *             If the value represented is zero.
+     */
+    public int getOrigMagnitude() throws ArithmeticException;
```

Comment:
How about `getExponent()` instead?  And it need not throw ArithmeticException if you just return the exponent (which you can say is well-defined for zero).

Reply:
Done.